components/InputReducer/InputReducer.js

```diff
@@ -0,0 +1,39 @@
+export default function InputReducer(state, action) {
+  if (action.type === "input") {
+    return {
+      ...state,
+      [action.name]: action.value,
+    }
+  } else if (action.type === "object-input") {
+    //Make sure to send objectVariables prop to populate object with current key/value pairs.
+    return {
+      ...state,
+      [action.object]: {
+        ...action.objectVariables,
+        [action.name]: action.value,
+      },
+    }
```

Comment:
This one was a little tricky. Since the social sites are stored as an object, I had to come up with this logic in order to make sure that the existing social sites are also included. Might be easier to explain via video.